### PR TITLE
[IMP] [stock_picking_transfer_lot_autoassign] Avoid auto-assignment of lots based on picking types

### PR DESCRIPTION
**Pull request by cubells**
This allows lot administrator to avoid auto-assignment of lots depending on picking type.

cc @Tecnativa

### CODE REVIEW

File: stock_picking_transfer_lot_autoassign/models/stock_picking.py [hunk 1]
```diff
@@ -14,11 +14,20 @@ def _prepare_pack_ops(self, picking, quants, forced_qties):
...
             picking, quants, forced_qties,
         )
-        for pack_op_vals in res:
-            qty_done = 0
-            for pack_lot_vals in pack_op_vals.get('pack_lot_ids', []):
-                pack_lot_vals[2]['qty'] = pack_lot_vals[2]['qty_todo']
-                qty_done += pack_lot_vals[2]['qty_todo']
-            if qty_done:
-                pack_op_vals['qty_done'] = qty_done
+        if not picking.picking_type_id.avoid_internal_assignment:
```
Comment (pedrobaeza) [inline stock_picking_transfer_lot_autoassign/models/stock_picking.py]: Better to not indent all the block:
```
if picking.picking_type_id.avoid_internal_assignment:
    return res
for ...
```

File: stock_picking_transfer_lot_autoassign/models/stock_picking.py [hunk 2]
```diff
@@ -14,11 +14,20 @@ def _prepare_pack_ops(self, picking, quants, forced_qties):
...
+                qty_done = 0
+                for pack_lot_vals in pack_op_vals.get('pack_lot_ids', []):
+                    pack_lot_vals[2]['qty'] = pack_lot_vals[2]['qty_todo']
+                    qty_done += pack_lot_vals[2]['qty_todo']
+                if qty_done:
+                    pack_op_vals['qty_done'] = qty_done
         return res
+
+
+class StockPickingType(models.Model):
```
Comment (pedrobaeza) [inline stock_picking_transfer_lot_autoassign/models/stock_picking.py]: Separated file

File: stock_picking_transfer_lot_autoassign/views/stock_picking_view.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (pedrobaeza) [inline stock_picking_transfer_lot_autoassign/views/stock_picking_view.xml]: stock_picking_type_view.xml

**Review by pedrobaeza — APPROVED** [review]
Squashing on merging

**Comment by luismontalba** [comment]
I create a stockable product and mark the "tracking by lots" feature.
I edit the picking type form for delivery orders marking the option "Avoid auto-assigment of lots":

seleccion_001

I receipt (100+100+100) units of the product assigning lot numbers L001, L002 and L003:

seleccion_002

I create a delivery order for 50 units of the product, but the lot number L001 (the first created) is auto-assigned:

seleccion_003

Is this the expected result?

**Comment by pedrobaeza** [comment]
Yeah, absolutely. The idea is to avoid the auto-assignation when checked. You can try the contrary, not checking, and see the lines green.

**Review by luismontalba — APPROVED** [review]
OK, tested on runbot

**Pull request merged**